### ISSUE TITLE
Will we create our own backend or use a service.

### ISSUE THREAD

**Issue by Blaze349** [gitscratch/gitscratch#7]
We may need a backend to store:
* Users and login details
* Git projects and permissions

If we use a backend then we will need hosting such as 
* Heroku (free)
* OpenShift (free)
* AWS (requires credit card) (free)

It seems like the intention, though is to host this on GitHub pages. Github is completely static so we cannot use it alone for the project.

We could use a service such as Firebase to store our user logins, what projects they have and other things.
We could use github to store the actual projects.

This has two limitations though. 
* The User must have a github account as well as a scratch account.
* Private collaboration requires money.
* How comfortable are we with giving google data?

Here are some links:
* [Heroku](https://www.heroku.com/)
* [Amazon Web Services](https://aws.amazon.com/)
* [OpenShift by Red Hat](https://www.openshift.com/)
* [Firebase](https://firebase.google.com/)

**Comment by card100** [comment]
The best I have is a rpi

**Comment by Blaze349** [comment]
Raspberry pi? It could work but it would be impractical.

**Comment by card100** [comment]
Ik
We're going to need something better.

**Comment by Blaze349** [comment]
Maybe HeroHamp knows a good host

**Comment by herohamp** [comment]
the 11 months digitalocean gives you for free?

**Comment by herohamp** [comment]
https://education.github.com/pack

**Comment by Blaze349** [comment]
That seems good but what will happen after 11 months? The project has no form of revenue

**Comment by herohamp** [comment]
I dunno. I think we should just make a something that uses github. Like a app (downloaded) that you select the sb2 file log into your github and that program would upload

https://electron.atom.io/ ?

**Comment by Blaze349** [comment]
Wouldn't it be easier to make a web app that grabs the sb2 file directly from scratch then uploads it to github? Or a chrome/firefox extension?

I would have liked for us to make a tosh wrapper with git. So that you could git pull and push directly from tosh.

**Comment by herohamp** [comment]
Well the web app would not be able to upload to git. Right? but the downloaded could use ANY node package made to upload to github.

**Comment by Blaze349** [comment]
It should be able to, using the git api.
Since the web app is running on node.js it can use whatever node.js package that exists

**Comment by herohamp** [comment]
Blaze but a web app would require hosting. If we just used a app then we could just use github pages to distribute it

**Comment by Blaze349** [comment]
That's true...But the problem is that many scratcher's cannot download apps...
We don't actually need hosting though.

Since we are using github to store everything the only then we don't need hosting at all.
We can create a static page that grabs an sb2 file and commits and pushes it to git.
The thing is we only need hosting for user authentication but firebase handles user authentication for free without hosting

**Comment by herohamp** [comment]
Ummmm... javascript cant push github

**Comment by herohamp** [comment]
How about we just talk on slack. Whats you email? I made gitscratch a room

**Comment by Blaze349** [comment]
You can actually. But you have to use the github api. My email is blaze349@protonmail.com

Could we use gitter.im instead. it doesn't requires signup only a github account

**Comment by herohamp** [comment]
Ok

**Comment by herohamp** [comment]
Hmmm can you make a demo using the github api? and just javascript?

**Comment by Blaze349** [comment]
Ok. But I'm going to Jerusalem tomorrow morning so I will try to do it as soon as possible

**Comment by herohamp** [comment]
ok. Cool btw!

**Comment by herohamp** [comment]
https://gitter.im/gitscratch-/Lobby#

**Comment by Blaze349** [comment]
Thanks for setting it up!

**Comment by ALinuxLover** [comment]
https://developer.github.com
???

**Comment by Blaze349** [comment]
This is a better wrapper:
http://www.nodegit.org

@iamunknown2 created a perl file that could handle it. The only problems are speed, async and integration with node.js

**Comment by ALinuxLover** [comment]
ok
